recipe/meta.yaml

```diff
@@ -28,12 +28,10 @@ requirements:
     - cf_units
     - dask >=0.17.1
     - matplotlib
-    - netcdf4 <1.4
```

Comment:
I'd love to see this change in upstream's CI first. I have a branch (I'll commit it soon, I was waiting on https://github.com/SciTools/iris/pull/3016 which is blocked b/c of https://github.com/SciTools/nc-time-axis/issues/35) that show some errors when using `netcdf4 >=1.4` with iris.